test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -371,7 +378,7 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 	if err != nil {
 		return ctrl.Result{}, errors.Wrap(err, "failed to generate workload cluster client")
 	}
-	if err := externalMachine.SetNodeProviderID(ctx, remoteClient); err != nil {
+	if err := externalMachine.CloudProviderNodePatch(ctx, remoteClient, dockerMachine); err != nil {
```

Comment:
I assume we have to do something similar for docker machine pools at https://github.com/kubernetes-sigs/cluster-api/blob/38653d8c4b4b76c90fe8b1e2ce1f47397af06c78/test/infrastructure/docker/exp/internal/docker/nodepool.go#L345

Reply:
We don't if we don't support setting external cloud provider for docker machine pools. I would put off doing this (and create a follow up issue if you're happy with that) until we have DockerPool machines in place.